ui/pages/confirmations/confirmation/useConfirmationRouting.ts

```diff
@@ -0,0 +1,25 @@
+import { useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { isEqual } from 'lodash';
+import { getUnapprovedTemplatedConfirmations } from '../../../selectors';
+
+/**
+ * A utility hook that parses an optional ID from the URL and
+ * returns the confirmation with that ID for routing purposes.
+ *
+ * @returns A pending confirmation if applicable.
+ */
+export function useConfirmationRouting() {
```

Comment:
Sorry for the confusion! When I suggested a hook I was meaning to abstract the `history.push` and building the path to a specific confirmation, so we didn't have to repeat that in future.

This logic looks great but as it's quite lean, would it be easier to follow within the original `confirmation.js` so we don't have to re-select the pending confirmations?

Reply:
Ah, sure! I will move it to the confirmation.js file